### PR TITLE
Remove `model_type` from the `RolePlaying`

### LINKED ISSUE #190: [Feature Request] Allow different models for `assistant`, `user`, `task_specify_agent`, and `critic` etc in the `RolePlaying`
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

As number of LLMs is increasing, many LLMs are designed for different tasks and even have different price. So, it's useful to let user specify different models for each agent in `RolePlaying`. Followings are some examples:

**Example 1**
User may have cost concern for `RolePlaying` when they use for example GPT4 for all `assistant`, `user` and `critic`. One way to lower cost might be using GPT3.5 for `assistant` and `user` but GPT4 for `critic`. In this way user can have some flexibility to control the cost of `RolePlaying` but still maintain some quality of `RolePlaying`.

**Example 2**
Now there are many LLMs are trained to optimize some specific tasks. In the future, it's possible that user can use a specific LLM for `assistant` but another LLM for `user` during `RolePlaying` collaboration. Maybe in this way the `RolePlaying` can have better performance.

### Solution

Allow different models for `assistant`, `user`, `task_specify_agent`, and `critic` etc in the `RolePlaying`

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zechengz**
## Description

Remove "global" `model_type` from the `RolePlaying`.
User can specify `model` in each `*agent_kwargs` instead, which is more flexible. Especially if user want to use different models for different agents during role playing.

This PR includes:
- Modification of `RolePlaying`
- Add / modify corresponding unit tests to `test/agents/test_role_playing.py`
- Modify current examples

## Motivation and Context

Fixes #190

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)
- [x] Example (update in the folder of example)

## Checklist

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: examples/ai_society/role_playing.py [hunk 1]
```diff
@@ -69,4 +72,4 @@ def main(model_type=None) -> None:
 
 
 if __name__ == "__main__":
-    main()
+    main(model_type=ModelType.GPT_3_5_TURBO_16K)
```
Comment (Obs01ete) [inline examples/ai_society/role_playing.py:75]: Why not going with None so that the default is used for all agents?

**Review by Obs01ete — COMMENTED** [review]
Overall great implementation!

File: examples/ai_society/role_playing.py (hunk 1 above)
Comment (zechengz) [reply examples/ai_society/role_playing.py]: I changed that to have a quick test on examples 😂 Just revert the change

**Review by Obs01ete — COMMENTED** [review]
I will approve this soon, once I fix the tests.

**Review by Obs01ete — APPROVED** [review]

**Comment by lightaime** [comment]
I think it is better to have still a global `model_type` as default. Only change the `model_type` for agents that are specified using `*agent_kwargs` what do you think @zechengz @Obs01ete?

**Comment by zechengz** [comment]
@lightaime IMO it depends on whether we should provide an easy way for user to specify a type of model to all agents directly.
In my opinion in the future we should pass initialized agents to role playing instead of passing agent name and its `kwargs` etc. Especially when we have more and complex agents in role playing (also passing agents directly may enable user to create their own types of agents much easier). Now we bind each agent with a model directly, so before using role playing (before passing agents to role playing), user should already have ideas on the model they use for each agent.
If we really want to have a "global" `model_type` to override, we may make it as optional and set the default value as `None`. Only if user specifies the `model_type`, we will use it to replace the model in all agents. In documentation we can say that this `model_type` is aimed to provide a "shortcut" to replace or override the model for all agents.
Though in this case this might be a little bit duplicated with the `model_type` specified for each agent (if you agreed that we should pass initialized agents to role playing directly in the future)
Also, now both agent default model and previous role playing default `model_type` is gpt3.5. So in the short term there is no much influence.
WDYT?

**Comment by lightaime** [comment]
@zechengz, yes, in the future we will allow users to pass initialized agents to the API directly. We may support this with another API instead of `RolePlaying`. So no worries. For the `RolePlaying` API, I think it is better to keep it easy to use since most of the time we use the same `model_type` for all the agents. An argument to set the `model_type` for all the agents would be handy. We can rename `model_type` as `default_model_type` and do something like `*_agent_kwargs.setdefault(model, default=default_model_type)`. WDYT?

**Comment by zechengz** [comment]
@lightaime I see, I can help to add that.

**Pull request merged**